cmd/create/cluster/cmd.go

```diff
@@ -2251,6 +2257,42 @@ func run(cmd *cobra.Command, _ []string) {
 	}
 }
 
+func validateOperatorRolesAvailabilityUnderUserAwsAccount(awsClient aws.Client,
+	operatorIAMRoleList []ocm.OperatorIAMRole) error {
+	for _, role := range operatorIAMRoleList {
+		name, err := aws.GetResourceIdFromARN(role.RoleARN)
+		if err != nil {
+			return err
+		}
+
+		err = awsClient.ValidateRoleNameAvailable(name)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func validateOperatorRolesMatchOidcProvider(awsClient aws.Client,
+	operatorIAMRoleList []ocm.OperatorIAMRole, oidcEndpointUrl string) error {
+	operatorIAMRoles := operatorIAMRoleList
+	for _, operatorIAMRole := range operatorIAMRoles {
+		roleARN := operatorIAMRole.RoleARN
+		roleObject, err := awsClient.GetRoleByARN(roleARN)
+		if err != nil {
+			return err
+		}
+		parsedUrl, err := url.Parse(oidcEndpointUrl)
```

Comment:
Can we extract the parsing outside of the loop?

Reply:
Sure, I feel like I did this lol. Maybe it was in CS or it got left out in some stash